### PR TITLE
[gym] add result_bundle_path option.

### PR DESCRIPTION
**Pull request by barakwei**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
If you want to use a result bundle with `gym`, your only option is to set `result_bundle` to `true`, and this will create a result bundle in the output directory with the path `#{output_name}.result`. Currently, it's not possible to have the result bundle outside the output directory to to choose its path.

If you want the path of the created result bundle you could either create this on your own by manually joining the directories and `.result`, or by using the cache (`Gym.cache[:result_bundle_path]`). Both solutions are implementation details and therefore suboptimal.

Related PR: #12357

### Description
I propose to add a new option - `result_bundle_path`, which is active only when `result_bundle` is `true`. This will create the result bundle in the provided path.

### CODE REVIEW

**Comment by janpio** [comment]
Tests fail on AppVeyor/Windows:

```
Failures:
  1) Gym Gym::BuildCommandGenerator Result Bundle Path Example does not use result_bundle_path if result_bundle is false
     Failure/Error: Gym.config = FastlaneCore::Configuration.create(Gym::Options.available_options, options)
     FastlaneCore::Interface::FastlaneError:
       Unable to locate Xcode. Please make sure to have Xcode installed on your machine
     # ./fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!'
     # ./fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
     # ./fastlane_core/lib/fastlane_core/helper.rb:166:in `xcode_at_least?'
     # ./fastlane_core/lib/fastlane_core/project.rb:329:in `build_xcodebuild_showbuildsettings_command'
     # ./fastlane_core/lib/fastlane_core/project.rb:350:in `build_settings'
     # ./fastlane_core/lib/fastlane_core/project.rb:293:in `supported_platforms'
     # ./fastlane_core/lib/fastlane_core/project.rb:281:in `mac?'
     # ./gym/lib/gym/detect_values.rb:114:in `detect_platform'
     # ./gym/lib/gym/detect_values.rb:32:in `set_additional_default_values'
     # ./gym/lib/gym/module.rb:15:in `config='
     # ./gym/spec/build_command_generator_spec.rb:227:in `block (4 levels) in <top (required)>'
Finished in 2 minutes 39 seconds (files took 4.41 seconds to load)
5632 examples, 1 failure, 184 pending
```

Annotating the spec with `, requires_xcode: true` or `, requires_xcodebuild: true` should fix this.

(I also restarted the other tests because the Linux failed because of unrelated reasons)

**Comment by barakwei** [comment]
Thanks @janpio ,this solved it for Windows, but it looks like the linux job still has some unrelated problem.

**Comment by janpio** [comment]
Correct, that seems to be a networking or CDN problem with the Debian mirrors :/ Right now I hope this will fix itself soon - so we can fully ignore this here for now.

**Comment by joshdholtz** [comment]
@barakwei Can you rebase from master when you get a chance or give contributors write access to your fork? 😊 Once this gets rebased we can get this merged!

**Comment by barakwei** [comment]
rebased

**Comment by barakwei** [comment]
@janpio @joshdholtz can this be merged?

**Comment by janpio** [comment]
Someone of the team will have to be confident enough in the changes to leave an Approve review and/or merge it. For me, that usually means actually testing some code locally, which takes an appropriate amount of time to set up and test - which I currently don't have.

**Comment by barakwei** [comment]
@janpio @joshdholtz bump

**Comment by barakwei** [comment]
@janpio @joshdholtz bump

File: gym/lib/gym/options.rb
Comment (janpio) [inline gym/lib/gym/options.rb]: any specific motivation for choosing `e`?

File: gym/lib/gym/generators/build_command_generator.rb
Comment (janpio) [inline gym/lib/gym/generators/build_command_generator.rb]: Does putting the `.result` inside here make any real difference or is this just nicer or more consistent with how `File.join` should be used?

**Review by janpio — APPROVED** [review]
Outside of my area of expertise, but change looks safe enough if the new option is not used, so I am fine with reviewing this and later merge it. 

Left 2 probably silly comments, when those are addressed this can be merged.

File: gym/lib/gym/options.rb
Comment (barakwei) [reply gym/lib/gym/options.rb]: no real reason, it's the only first letter that was unused, I can remove it if needed.

File: gym/lib/gym/generators/build_command_generator.rb
Comment (barakwei) [reply gym/lib/gym/generators/build_command_generator.rb]: I'm not sure what exactly are you referring to, the result directory name should be `#{output_dir}/#{output_name}.result`. `File.join` adds the file separator between all the parameters. I couldn't do `File.join(our_dir, out_name, ".result)` as that would create `#{output_dir}/#{output_name}/.result`

File: gym/lib/gym/options.rb
Comment (janpio) [reply gym/lib/gym/options.rb]: I actually don't know how the short options are selected. @joshdholtz?

**Comment by barakwei** [comment]
@joshdholtz can you give this a quick looks pls?

**Comment by barakwei** [comment]
@janpio I removed the short options as it's not important

**Comment by janpio** [comment]
Thanks, finally merged 🚀

**Comment by barakwei** [comment]
Thanks @janpio!!!

**Pull request merged**
